v2/contribute/decisions/oauth/0011-frontend-routes.mdx

```diff
@@ -0,0 +1,48 @@
+---
+id: "0011"
+title: Redirect to the standard login route if required
+hide_title: true
+---
+
+import DecisionHeader from "/src/components/decisionLogs/DecisionHeader"
+import ArgumentPro from "/src/components/decisionLogs/ArgumentPro"
+import ArgumentCon from "/src/components/decisionLogs/ArgumentCon"
+import ArgumentNeut from "/src/components/decisionLogs/ArgumentNeut"
+
+# Redirect to the standard login route if required
+
+<DecisionHeader status="proposed" lastUpdate="2023-05-11" created="2023-05-11" deciders={["rishabhpoddar", "porcellus"]} proposedBy={["porcellus"]} />
+
+## Context and Problem Statement
+
+When the user redirected to the app during the OAuth2 sign-in process, there are a few routes we can choose to provide.
+
+## Considered Options
+
```

Comment:
releated to the frontend is also - how will we store the scopes, redirect_uri etc from the start of the flow till the end? Cause logging in can be a multi step process involving email verification, 2fa etc.. We also need to keep in mind that something like passwordless login will end up creating a session in the other browser (if the link is opened on another browser), so in this case, how will the scopes, redirect uri be transferred?

Reply:
1. I'd say we should store them in sessionstorage (localstorage is also an option):
    - Navigating inside the page (and waiting for an email to be verified for example) works
    - No security issues with someone starting a login process, closing the tab/browser then you opening it and logging into something unintentionally.
    - If the email was verified in another tab/device we can continue the flow after a refresh (we may want to implement some kind of polling for this to not require refreshing)
2. We can't really continue the login process on another device (or at least it would not be standard): 
    - even if we added all info into the magic link (or transfered it to the other device through other means) the state would be missing from the client after the redirection
    - this also matches the general magic link flow we have - the original tab keeps checking for an active session then goes into a "You have been successfully signed in. Please Close this tab" state (which will have to change to check if it needs to do this success redirect)

Reply:
So this breaks the passwordless login flow that we have - cause when the magic link is clicked and opens in a different browser (not tab), the session is only created in the other browser. The original browser doesn't know about the session creation and is in the same page as "check your email". The way things are right now, passwordless login via magic links won't work for oauth 2.

This probably also explains why all other providers log the user into the original browser and not the new browser (this was a question we were pondering over at the time we were implementing passwordless login).

Ideally, i would not like to change the passwordless magic link flow now (cause that requires work etc..), but instead, we could disallow people from using magic link based login when using in the context of an oauth 2 provider. What do you think?

Reply:
Also, using session storage may not be the best idea cause it's only per tab. If the user goes through the email verification flow, and opens the link in a new tab (same browser), then the session storage on that tab will not have the right info - breaking the oauth flow.